core/lib/zksync_core/src/api_server/execution_sandbox/vm_metrics.rs

```diff
@@ -1,104 +1,189 @@
+    #[metrics(buckets = MEMORY_SIZE_BUCKETS)]
+    memory_size: Family<SizeType, Histogram<usize>>,
+    #[metrics(buckets = MEMORY_SIZE_BUCKETS)]
+    decommitter_size: Family<SizeType, Histogram<usize>>,
+    #[metrics(buckets = MEMORY_SIZE_BUCKETS)]
+    storage_size: Family<SizeType, Histogram<usize>>,
+    #[metrics(buckets = MEMORY_SIZE_BUCKETS)]
+    storage_view_cache_size: Histogram<usize>,
+    #[metrics(buckets = MEMORY_SIZE_BUCKETS)]
+    full: Histogram<usize>,
+}
+
+#[vise::register]
+static MEMORY_METRICS: vise::Global<RuntimeContextMemoryMetrics> = vise::Global::new();
+
+const INTERACTION_AMOUNT_BUCKETS: Buckets = Buckets::exponential(10.0..=10_000_000.0, 10.0);
+
+#[derive(Debug, Metrics)]
+#[metrics(prefix = "runtime_context_storage_interaction")]
+struct RuntimeContextStorageMetrics {
+    #[metrics(buckets = INTERACTION_AMOUNT_BUCKETS)]
+    amount: Family<InteractionType, Histogram<usize>>,
+    #[metrics(buckets = Buckets::LATENCIES)]
+    duration: Family<InteractionType, Histogram<Duration>>,
+    #[metrics(buckets = Buckets::LATENCIES)] // FIXME: probably should use smaller buckets
+    duration_per_unit: Family<InteractionType, Histogram<Duration>>,
+    #[metrics(buckets = Buckets::ZERO_TO_ONE)]
+    ratio: Histogram<f64>,
+}
+
+#[vise::register]
+static STORAGE_METRICS: vise::Global<RuntimeContextStorageMetrics> = vise::Global::new();
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, EncodeLabelValue, EncodeLabelSet)]
+#[metrics(label = "stage", rename_all = "snake_case")]
+pub(super) enum SandboxStage {
+    VmConcurrencyLimiterAcquire,
+    Initialization,
+    ValidateInSandbox,
+    Validation,
+    Execution,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, EncodeLabelValue, EncodeLabelSet)]
+#[metrics(label = "stage", rename_all = "snake_case")]
+pub(in crate::api_server) enum SubmitTxStage {
+    #[metrics(name = "1_validate")]
+    Validate,
+    #[metrics(name = "2_dry_run")]
+    DryRun,
+    #[metrics(name = "3_verify_execute")]
+    VerifyExecute,
+    #[metrics(name = "4_tx_proxy")]
+    TxProxy,
+    #[metrics(name = "4_db_insert")]
+    DbInsert,
+}
+
+#[derive(Debug, Metrics)]
+#[metrics(prefix = "api_web3")]
+pub(in crate::api_server) struct SandboxMetrics {
+    #[metrics(buckets = Buckets::LATENCIES)]
+    pub(super) sandbox: Family<SandboxStage, Histogram<Duration>>,
+    #[metrics(buckets = Buckets::linear(0.0..=2_000.0, 200.0))]
+    pub(super) sandbox_execution_permits: Histogram<usize>,
+    #[metrics(buckets = Buckets::LATENCIES)]
+    pub submit_tx: Family<SubmitTxStage, Histogram<Duration>>,
+    #[metrics(buckets = Buckets::linear(0.0..=30.0, 3.0))]
+    pub estimate_gas_binary_search_iterations: Histogram<usize>,
+}
+
+#[vise::register]
+pub(in crate::api_server) static SANDBOX_METRICS: vise::Global<SandboxMetrics> =
+    vise::Global::new();
+
+#[derive(Debug, Metrics)]
+#[metrics(prefix = "api_execution")]
+pub(super) struct ExecutionMetrics {
```

Comment:
These metrics look like they should be a part of `SandboxMetrics`, but they have a differing prefix (`api_execution_*` vs `api_web3_*`).